### ISSUE TITLE
Getting 'TypeError: Cannot read property 'get' of undefined' when trying to connect

### ISSUE THREAD

**Issue by briandela** [hapijs/reptile#14]
Trying to use reptile with Hapi 7. I add the plugin and it registers correctly (I'm using the default settings). However, when I try to connect I get the following error `TypeError: Cannot read property 'get' of undefined` from this line `var context = Hoek.applyToDefaults({ socket: socket, pack: pack }, options.context || {});`

This is the extended stacktrace (from `longjohn`):

```
/Users/brian/farmstr/website/node_modules/longjohn/dist/longjohn.js:185
        throw e;
              ^
TypeError: Cannot read property 'get' of undefined
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:67:31)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.clone (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:71:41)
    at exports.applyToDefaults (/Users/brian/farmstr/website/node_modules/hoek/lib/index.js:152:24)
---------------------------------------------
    at new Server (net.js:923:10)
    at exports.createServer (net.js:75:10)
    at exports.register (/Users/brian/farmstr/website/node_modules/reptile/lib/index.js:22:22)
    at internals.Pack._plugin (/Users/brian/farmstr/website/node_modules/hapi/lib/pack.js:455:21)
    at /Users/brian/farmstr/website/node_modules/hapi/lib/pack.js:261:14
    at iterate (/Users/brian/farmstr/website/node_modules/hapi/node_modules/items/lib/index.js:35:13)
    at done (/Users/brian/farmstr/website/node_modules/hapi/node_modules/items/lib/index.js:27:25)
    at /Users/brian/farmstr/website/node_modules/tv/lib/index.js:117:16
---------------------------------------------
    at internals.Server._init (/Users/brian/farmstr/website/node_modules/hapi/lib/server.js:344:19)
    at internals.Server._start (/Users/brian/farmstr/website/node_modules/hapi/lib/server.js:304:10)
    at /Users/brian/farmstr/website/node_modules/hapi/lib/pack.js:564:14
    at iterate (/Users/brian/farmstr/website/node_modules/hapi/node_modules/items/lib/index.js:35:13)
    at exports.serial (/Users/brian/farmstr/website/node_modules/hapi/node_modules/items/lib/index.js:38:9)
    at internals.Pack._apply (/Users/brian/farmstr/website/node_modules/hapi/lib/pack.js:562:11)
    at /Users/brian/farmstr/website/node_modules/hapi/lib/pack.js:530:18
    at /Users/brian/farmstr/website/node_modules/hoek/lib/index.js:642:22
```

Any ideas?

**Comment by briandela** [comment]
@geek I ended up tracking down the issue. Essentially, one of my plugins has an options object which looks like this:
```js
options:  {
    env: process.env
}
```

In this case, when `var context = Hoek.applyToDefaults({ socket: socket, pack: pack }, options.context || {});` (https://github.com/hapijs/reptile/blob/master/lib/index.js#L29) is called, Hoek will try to do a deep clone and ends up crashing when trying to clone the `options.env` of the plugin.

The issue appears to be in Hoek here https://github.com/hapijs/hoek/blob/master/lib/index.js#L67. There is no check for decriptor being undefined in this code:

```js
var descriptor = Object.getOwnPropertyDescriptor(obj, i);
if (descriptor.get ||
    descriptor.set) {

    Object.defineProperty(newObj, i, descriptor);
}
else {
    newObj[i] = exports.clone(obj[i], seen);
}
```

`Object.getOwnPropertyDescriptor` can return undefined and `descriptor.get` then throws.

I've worked around the issue by just not passing in process.env as an option to the plugin (not sure why that was done in the first place rather than just calling `process.env` inside the plugin)
